schema/users.ts

```diff
@@ -124,5 +153,14 @@ export default {
     lastActiveAt: { type: 'date' },
 
     blockedReason: { type: 'keyword' },
+
+    badges: {
+      type: 'nested',
+      properties: {
+        badgeId: { type: 'keyword' },
+        badgeMetaData: { type: 'keyword' },
+        isDisplay: { type: 'boolean' }
```

Comment:
Missing `createdAt` and `updatedAt`. I think it's good for the nested objects of `badge` to have these timestamps, recording the timestamp of when the user is granted a badge, and maybe the last time its metadata is modified using Admin API by Badge managers.